addons/mail/models/mail_thread.py

```diff
@@ -2608,7 +2608,7 @@ def _notify_classify_recipients(self, recipient_data, model_name, msg_vals=None)
         for group_name, group_func, group_data in groups:
             group_data.setdefault('notification_group_name', group_name)
             group_data.setdefault('notification_is_customer', False)
-            group_data.setdefault('has_button_access', True)
+            group_data.setdefault('has_button_access', bool(msg_vals and msg_vals.get('model') and msg_vals.get('res_id')))
```

Comment:
We should probably support a true model-based call, like ``_notify_get_action_link`` that uses
```
        params = {
            'model': kwargs.get('model', self._name),
            'res_id': kwargs.get('res_id', self.ids and self.ids[0] or False),
        }
```

One thing that could be done is therefore to compute something like
``is_thread_notification = kwargs.get('model', self._name) != 'mail.thread' and len(kwargs.get('res_id', self.ids))`` that indicate we are calling the method on a real record. Would that do the trick ?

Reply:
Updated, res_id is actually an integer, a bit of marshmalling will be necessary :)